pkgs/applications/networking/mailreaders/nylas-mail/default.nix

```diff
@@ -0,0 +1,137 @@
+{ dpkg, fetchurl, lib, pkgs, stdenv, config
+, alsaLib
+, atk
+, cairo
+, coreutils
+, cups
+, dbus
+, desktop_file_utils
+, expat
+, fontconfig
+, freetype
+, gcc-unwrapped
+, gdk_pixbuf
+, glib
+, gnome2
+, libgcrypt
+, libgnome_keyring
+, libnotify
+, makeWrapper
+, nodejs
+, nspr
+, nss
+, pango
+, python2
+, udev
+, wget
+, xorg
+}:
+
+stdenv.mkDerivation rec {
+  name = "${pkgname}-${version}";
+  pkgname = "nylas-mail";
+  version = "2.0.32";
+  subVersion = "fec7941";
+
+  src =
+    if stdenv.system == "x86_64-linux" then
+      fetchurl {
+        url = "https://edgehill.s3-us-west-2.amazonaws.com/${version}-${subVersion}/linux-deb/x64/NylasMail.deb";
+        sha256 = "40060aa1dc3b5187b8ed4a07b9de3427e3c5a291df98c2c82395647fa2aa4ada";
+      }
+    else
+      throw "NylasMail is not supported on ${stdenv.system}";
+
+  # Build dependencies
+  propagatedBuildInputs = [
+    alsaLib
+    atk
+    cairo
+    coreutils
+    cups
+    dbus
+    desktop_file_utils
+    expat
+    fontconfig
+    freetype
+    gcc-unwrapped
+    gdk_pixbuf
+    glib
+    gnome2.GConf
+    gnome2.gtk
+    libgnome_keyring
+    libnotify
+    nodejs
+    nspr
+    nss
+    pango
+    python2
+    udev
+    wget
+    xorg.libX11
```

Comment:
@joachifm If you were talking about the `lib.makeLibraryPath` section I could do [this](https://github.com/johnramsden/NixOS/commit/159f14aaf36043714315110f0780ee690669a621), I just passed in the `propagatedBuildInputs` because they were the right packages.